lambda/src/new_account_iam_role.py

```diff
@@ -0,0 +1,383 @@
+    if not iam_role_create_policy(iam_client, role, role_name, policy_arn):
+        raise IamRoleInvalidArgumentsError(
+            f"Unable to attach '{policy_arn}'to {role_name}."
+        )
+    return 0
+
+
+def lambda_handler(event, context):  # pylint: disable=unused-argument
+    """Entry point if script called by AWS Lamdba."""
+    LOG.info("Received event:\n%s", json.dumps(event))
+
+    # For the CLI entrypoint (main), argparse will ensure that required
+    # arguments are provided.  The Lambda entrypoint uses environment
+    # variables to supply "arguments" and checks for required arguments
+    # are performed here.
+
+    # Optional:  Used to create assume-role-arn.
+    assume_role_name = os.environ.get("ASSUME_ROLE_NAME")
+
+    # Required:  Used for role-name, e.g., E_READONLY or E_PROVADMIN or
+    # E_PROVREADONLY.
+    role_name = os.environ.get("ROLE_NAME")
+
+    # Required:  role-permission-policy.  AWS-managed permission policy to
+    # attach to the role.
+    permission_policy = os.environ.get("PERMISSION_POLICY")
+
+    # Required:  trust-policy-json.  JSON-formatted string with trust policy.
+    trust_policy_json = os.environ.get("TRUST_POLICY_JSON")
+
+    LOG.info(
+        "Environment variables:\n\tASSUME_ROLE_NAME=%s\n\tROLE_NAME=%s"
+        "\n\tPERMISSION_POLICY=%s\n\tTRUST_POLICY_JSON=%s",
+        assume_role_name,
+        role_name,
+        permission_policy,
+        trust_policy_json,
+    )
+
+    # Check for missing requirement environment variables.
+    msg = []
+    if not role_name:
+        msg.append(
+            "Environment variable 'ROLE_NAME' must provide "
+            "the name of the IAM role to create."
+        )
+    if not permission_policy:
+        msg.append(
+            "Environment variable 'PERMISSION_POLICY' must provide "
+            "the AWS-managed permission policy to attach to role."
+        )
+    if not trust_policy_json:
+        msg.append(
+            "Environment variable 'TRUST_POLICY_JSON' must be a JSON-"
+            "formatted string containing the role trust policy."
+        )
+    if msg:
+        LOG.error("\n".join(msg))
+        raise IamRoleInvalidArgumentsError("\n".join(msg))
+
+    # If there is no default boto cache dir use "/tmp/" as it is writable.
+    botocore_cache_dir = BOTOCORE_CACHE_DIR or "/tmp/.aws/boto/cache"
+
+    try:
+        account_id = get_account_id(event)
+        partition = get_partition()
+        role_arn = f"arn:{partition}:iam::{account_id}:role/{assume_role_name}"
+
+        main(
+            aws_profile=None,
+            role_name=role_name,
+            role_permission_policy=permission_policy,
+            trust_policy_json=trust_policy_json,
+            assume_role_arn=role_arn,
+            partition=partition,
+            botocore_cache_dir=botocore_cache_dir,
+        )
+    except (IamRoleInvalidArgumentsError, Exception) as exc:
```

Comment:
Actually, I should add AccountCreationFailedError to this list as well.  Regarding your question of duplication - yes, it will print the appropriate error regardless of whether I list IamRoleInvalidArgumentError, but it helps documents the type of errors I can expect.  It would be best if I didn't list Exception, but I'm worried there might be a situation that's totally unexpected and I don't want the code to crash.